docs/adr/0007-modelling-history.md

```diff
@@ -0,0 +1,272 @@
+# 7. Modelling history
+
+Date: 2018-12-17
+
+## Context
+
+Using terminology as defined in [ADR-3](../0003-initial-domain-modelling.md):
+
+- **Document**: A piece of content in a particular locale that can have many
+versions;
+- **Revision**: A particular edit of a piece of content, represents a document at
+a specific point in time;
+- **Edition**: A particular revision that was published to GOV.UK or is the
+current draft of the document.
+
+Content Publisher currently has a concept of a document and not editions or
+revisions. A consequence of this is that an edit to a document overwrites the
+previous state of the document. Since a document can exist in two forms, draft
+and live, and Content Publisher only knows of one we hit problems when
+knowledge of both would be useful.
+
+The suggestion is that we enhance Content Publisher to support the concepts of
+Document, Edition and Revision. The features that we'd like this to enable
+are:
+
+For a document with a live version and a draft:
+
+- The ability to discard the draft and revert the current version in Content
+  Publisher back to the live version (including all images and attachments) to
+  be in sync with Publishing API;
+- The ability to have the details of the live version, most noticeably state
+  and URL so this can be presented to users;
+- Ability to re-present either the live or draft version to Publishing API
+  should there be any problems with publishing.
+
+And with regards to history:
+
+- To know who did what to a document and when - effectively an audit trail of
+  accountability;
+- Be able to compare two revisions of a document to show changes;
+- Have stored in our database a full history of a document, which may go back
+  further than the Publishing API (Whitehall has longer histories of some
+  published documents)
+
+A [MoSCoW](https://en.wikipedia.org/wiki/MoSCoW_method) breakdown of this
+functionality was defined as:
+
+### Must
+
+- Store full information on latest and live versions of a document
```

Comment:
Is current essentially the draft?

Reply:
@emmabeynon yup that's correct.

It was renamed from "latest" to "current" because if your latest edition is a discarded then you want the live version to be the current one - if that makes sense?